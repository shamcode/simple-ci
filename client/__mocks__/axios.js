const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const { success, data } = FIXTURES[ method ][ url ];
            return Promise[ success ? 'resolve' : 'reject' ]( {
                data
            } );
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = ( () => {
    const data = {};
    data.chain = {
        id: 2,
        name: 'Test chain',
        command: 'echo 1'
    };
    data.project = {
        id: 1,
        name: 'Test',
        cwd: '/tmp/',
        chains: [ data.chain ]
    };
    return data;
} )();

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        return {
            ...MOCKS,
            interceptors: {
                request: {
                    use: () => {}
                },
                response: {
                    use: () => {}
                }
            }
        };
    } ),
    use( method, url, data, success = true ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"`);
        }
        FIXTURES[ method ][ url ] = {
            data,
            success
        };
        return this;
    },
    useDefaultMocks() {
        this
            .use( 'get', '/projects', [ defaultMocksData.project ] )
            .use( 'get', '/projects/1', defaultMocksData.project )
            .use( 'get',  '/chains/2', defaultMocksData.chain );
        return this;
    }
};