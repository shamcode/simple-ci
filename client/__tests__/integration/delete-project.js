import setup, { app } from './helpers'
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
    setup();
} );

it( 'delete project', async() => {
    expect.assertions( 5 );

    axios
        .useDefaultMocks()
        .use( 'delete', '/projects/1', null );

    await app.start();
    await app.project.open();

    app.click( '[href="projects/1/delete"]' );
    app.checkBody();
    await app.waitRendering();
    app.checkBody();

    await app.modal.ok();
    expect( axios.mocks.delete ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.delete.mock.calls[ 0 ][ 0 ] ).toBe( '/projects/1' );
    app.checkBody();
} );

