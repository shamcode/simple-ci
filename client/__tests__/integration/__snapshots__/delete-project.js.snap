// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete project 1`] = `
"<header>
  <a class=\\"logo\\" href=\\"\\"> Simple CI </a>
</header>
<!--0-->
<main>
  <ul class=\\"breadcrumb\\">
    <li>
      <a href=\\"\\"> Projects </a>
    </li>
    <!--0-->
    <li></li>
  </ul>
  <!--0-->
  <!--0-->
  <!--0-->
  <!--0-->
  <!--1-->
  <!--0-->
</main>
<footer> Sham 2018
  (v 0.0.7) </footer>
<!--1-->"
`;

exports[`delete project 2`] = `
"<header>
  <a class=\\"logo\\" href=\\"\\"> Simple CI </a>
</header>
<!--0-->
<main>
  <ul class=\\"breadcrumb\\">
    <li>
      <a href=\\"\\"> Projects </a>
    </li>
    <li>
      <a href=\\"projects/1\\">Test</a>
    </li>
    <!--0-->
    <!--0-->
    <li>Delete</li>
  </ul>
  <!--0-->
  <!--0-->
  <!--0-->
  <!--0-->
  <div class=\\"modal\\">
    <div class=\\"content\\">
      <div class=\\"header\\">Delete project</div>
      <div class=\\"body\\">Do you want delete project \\"Test\\"?</div>
      <div class=\\"footer\\">
        <button class=\\"ok danger\\"><span>Confirm</span></button>
        <button class=\\"cancel\\"><span>Cancel</span></button>
      </div>
    </div>
  </div>
  <!--0-->
  <!--1-->
  <!--0-->
  <!--1-->
  <!--0-->
</main>
<footer> Sham 2018
  (v 0.0.7) </footer>
<!--1-->"
`;

exports[`delete project 3`] = `
"<header>
  <a class=\\"logo\\" href=\\"\\"> Simple CI </a>
</header>
<!--0-->
<main>
  <!--0-->
  <!--0-->
  <!--0-->
  <div class=\\"project-list\\">
    <a class=\\"project-card\\" href=\\"projects/1\\"><span class=\\"name\\">Test</span></a>
    <!--0-->
    <a class=\\"project-card project-card-create\\" href=\\"projects/create\\">
      <div class=\\"create-icon\\"></div>
    </a>
  </div>
  <!--1-->
  <!--0-->
  <!--1-->
  <!--0-->
</main>
<footer> Sham 2018
  (v 0.0.7) </footer>
<!--1-->"
`;
